sktime/classification/dictionary_based/_tde.py

```diff
@@ -825,7 +825,8 @@ def _fit(self, X, y):
                     n_jobs=self._threads_to_use,
                 )
             )
-            sfa = self._transformers[0].fit_transform(X, y)
```

Comment:
Could you comment on why fit_transform does not work here?

Reply:
(or add a comment to the source code)

Reply:
SFA is not interface compliant, and adding the checks apparently breaks TDE.

Reply:
Ok, can you add this as a source code comment? Otherwise someone else might simply change it in a future PR.

Reply:
> SFA is not interface compliant, and adding the checks apparently breaks TDE.

I guess the problem might be with the input of a 2d numpy array to SFA, which are then treated as one multivariate time series. I have faced this issue before, too :(